mava/systems/sable/anakin/ff_sable.py

```diff
@@ -0,0 +1,668 @@
+    )
+    env_states, timesteps = jax.vmap(env.reset, in_axes=(0))(
+        jnp.stack(env_keys),
+    )
+    reshape_states = lambda x: x.reshape(
+        (n_devices, config.system.update_batch_size, config.arch.num_envs) + x.shape[1:]
+    )
+    # (devices, update batch size, num_envs, ...)
+    env_states = tree.map(reshape_states, env_states)
+    timesteps = tree.map(reshape_states, timesteps)
+
+    # Load model from checkpoint if specified.
+    if config.logger.checkpointing.load_model:
+        loaded_checkpoint = Checkpointer(
+            model_name=config.logger.system_name,
+            **config.logger.checkpointing.load_args,  # Other checkpoint args
+        )
+        # Restore the learner state from the checkpoint
+        restored_params, _ = loaded_checkpoint.restore_params(input_params=params)
+        # Update the params
+        params = restored_params
+
+    # Define params to be replicated across devices and batches.
+    key, step_keys = jax.random.split(key)
+    replicate_learner = (params, opt_state, step_keys)
+
+    # Duplicate learner for update_batch_size.
+    broadcast = lambda x: jnp.broadcast_to(x, (config.system.update_batch_size, *x.shape))
+    replicate_learner = tree.map(broadcast, replicate_learner)
+
+    # Duplicate learner across devices.
+    replicate_learner = flax.jax_utils.replicate(replicate_learner, devices=jax.devices())
+
+    # Initialise learner state.
+    params, opt_state, step_keys = replicate_learner
+
+    init_learner_state = LearnerState(
+        params=params,
+        opt_states=opt_state,
+        key=step_keys,
+        env_state=env_states,
+        timestep=timesteps,
+    )
+
+    return learn, eval_apply_fn, init_learner_state
+
+
+def run_experiment(_config: DictConfig) -> float:
+    """Runs experiment."""
+    config = copy.deepcopy(_config)
+
+    n_devices = len(jax.devices())
+
+    # Create the enviroments for train and eval.
+    env, eval_env = environments.make(config)
+
+    # PRNG keys.
+    key, key_e, net_key = jax.random.split(jax.random.PRNGKey(config.system.seed), num=3)
+
+    # Setup learner.
+    learn, sable_execution_fn, learner_state = learner_setup(env, (key, net_key), config)
+
+    # Setup evaluator.
+    def make_ff_sable_act_fn(actor_apply_fn: ActorApply) -> EvalActFn:
+        def eval_act_fn(
+            params: Params, timestep: TimeStep, key: chex.PRNGKey, actor_state: ActorState
+        ) -> Tuple[Action, Dict]:
+            output_action, _, _, _ = actor_apply_fn(  # type: ignore
+                params,
+                observation=timestep.observation,
+                key=key,
+            )
+            return output_action, {}
+
+        return eval_act_fn
+
+    # One key per device for evaluation.
+    eval_keys = jax.random.split(key_e, n_devices)
+    # Define Apply fn for evaluation.
+    # Create a fake hstate: will not be updated during steps
```

Comment:
```suggestion
    # Create an hstate with only zeros. This will never be updated over timesteps,
    # but will be updated between agents in a given timestep since ff_sable has no
    # memory over time. 
```